core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +561,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: (String) => Boolean = (s: String) => true): Option[FsApplicationAttemptInfo] = {
```

Comment:
oh ok - this internal replay helper method is called in two places in FsHistoryProvider. The two places  want different filtering - one wants all events replayed (call from `getAppUI()` ), the other (call from `mergeApplicationListing`) just the app start/end so they need to pass in the respective filters.